esphome/components/mcp4461/mcp4461.cpp

```diff
@@ -0,0 +1,662 @@
+      return 0;
+  }
+  if (nonvolatile) {
+    addr = addr + 0x20;
+  }
+  return addr;
+}
+
+uint16_t Mcp4461Component::get_wiper_level_(Mcp4461WiperIdx wiper) {
+  if (this->is_failed()) {
+    ESP_LOGE(TAG, "%s", LOG_STR_ARG(this->get_message_string(this->error_code_)));
+    return 0;
+  }
+  uint8_t wiper_idx = static_cast<uint8_t>(wiper);
+  if (!(this->reg_[wiper_idx].enabled)) {
+    ESP_LOGW(TAG, "%s", LOG_STR_ARG(this->get_message_string(MCP4461_WIPER_DISABLED)));
+    return 0;
+  }
+  if (!(this->reg_[wiper_idx].enabled)) {
+    ESP_LOGW(TAG, "reading from disabled volatile wiper %" PRIu8 ", returning 0", wiper_idx);
+    return 0;
+  }
+  return this->read_wiper_level_(wiper_idx);
+}
+
+uint16_t Mcp4461Component::read_wiper_level_(uint8_t wiper) {
+  uint8_t reg = 0;
+  uint16_t buf = 0;
+  reg |= this->get_wiper_address_(wiper);
+  reg |= static_cast<uint8_t>(Mcp4461Commands::READ);
+  if (wiper > 3) {
+    if (!this->is_eeprom_ready_for_writing_(true)) {
+      return 0;
+    }
+  }
+  if (!(this->read_byte_16(reg, &buf))) {
+    this->error_code_ = MCP4461_STATUS_I2C_ERROR;
+    this->status_set_warning();
+    ESP_LOGW(TAG, "Error fetching %swiper %" PRIu8 " value", (wiper > 3) ? "nonvolatile " : "", wiper);
+    return 0;
+  }
+  return buf;
+}
+
+bool Mcp4461Component::update_wiper_level_(Mcp4461WiperIdx wiper) {
+  if (this->is_failed()) {
+    ESP_LOGE(TAG, "%s", LOG_STR_ARG(this->get_message_string(this->error_code_)));
+    return false;
+  }
+  uint8_t wiper_idx = static_cast<uint8_t>(wiper);
+  if (!(this->reg_[wiper_idx].enabled)) {
+    ESP_LOGW(TAG, "%s", LOG_STR_ARG(this->get_message_string(MCP4461_WIPER_DISABLED)));
+    return false;
+  }
+  uint16_t data;
+  data = this->get_wiper_level_(wiper);
+  ESP_LOGV(TAG, "Got value %" PRIu16 " from wiper %" PRIu8, data, wiper_idx);
+  this->reg_[wiper_idx].state = data;
+  return true;
+}
+
+bool Mcp4461Component::set_wiper_level_(Mcp4461WiperIdx wiper, uint16_t value) {
+  if (this->is_failed()) {
+    ESP_LOGE(TAG, "%s", LOG_STR_ARG(this->get_message_string(this->error_code_)));
+    return false;
+  }
+  uint8_t wiper_idx = static_cast<uint8_t>(wiper);
+  if (value > 0x100) {
+    ESP_LOGW(TAG, "%s", LOG_STR_ARG(this->get_message_string(MCP4461_VALUE_INVALID)));
+    return false;
+  }
+  if (!(this->reg_[wiper_idx].enabled)) {
+    ESP_LOGW(TAG, "%s", LOG_STR_ARG(this->get_message_string(MCP4461_WIPER_DISABLED)));
+    return false;
+  }
+  if (this->reg_[wiper_idx].wiper_lock_active) {
+    ESP_LOGW(TAG, "%s", LOG_STR_ARG(this->get_message_string(MCP4461_WIPER_LOCKED)));
+    return false;
+  }
+  ESP_LOGV(TAG, "Setting MCP4461 wiper %" PRIu8 " to %" PRIu16 "!", wiper_idx, value);
+  this->reg_[wiper_idx].state = value;
+  this->reg_[wiper_idx].update_level = true;
+  return true;
+}
+
+void Mcp4461Component::write_wiper_level_(uint8_t wiper, uint16_t value) {
+  bool nonvolatile = false;
+  if (wiper > 3) {
+    nonvolatile = true;
+  }
```

Comment:
If you wanna get fancy... 😅 
```suggestion
  bool nonvolatile = wiper > 3;
```